src/shared/ask-password-api.c

```diff
@@ -862,10 +862,7 @@ int ask_password_agent(
                 char passphrase[LINE_MAX+1];
                 struct iovec iovec;
                 struct ucred *ucred;
-                union {
-                        struct cmsghdr cmsghdr;
-                        uint8_t buf[CMSG_SPACE(sizeof(struct ucred))];
-                } control;
+                CMSG_BUFFER_TYPE(CMSG_SPACE(sizeof(struct ucred))) control;
```

Comment:
We have `zero(control)` below. I think it would be better to initalize it here for consistency instead.

Reply:
given that this is about recvmsg the zero() should really be dropped without replacement